### PR TITLE
zone making functions

### PR DESCRIPTION
**Pull request by mem48**

### CODE REVIEW

**Comment by mem48** [comment]
@Robinlovelace FYI see new UI updates that come with this

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,4 @@ overline.*
 outputs/rnet/metadata.json
 outputs/routes/metadata.json
 outputdata/
+outputs/zones/metadata.json
```
Comment (Robinlovelace) [inline .gitignore:19]: Why exclude the metadata? Worth publishing in 'inputdata' if too big to commit or you don't want to share for other reasons. Important for reproducibility I think.

File: R/make_geojson.R [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#' Function to make geojson for tippecanoe (zones)
+#' @param z sf data frame with zones
+#' @param path folder to save results
+#' @example 
+# z_stats <- routes_to_zone_stats(r_fast, "fastest", "commute")
```
Comment (Robinlovelace) [inline R/make_geojson.R:5]: Good to see examples!

File: R/make_geojson.R [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# zones <- readRDS("../inputdata/iz_scotlands_uk.Rds")
```
Comment (Robinlovelace) [inline R/make_geojson.R:6]: We will need to tidy this up and maybe create example datasets with 

```
usethis::use_data()
```

But this is fine for now.

File: R/make_geojson.R [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+#           "dest_pcycle_godutch_fastest_commute",
+#           "orig_quietness_mean_fastest_commute",
+#           "orig_hilliness_mean_fastest_commute",
+#           "dest_quietness_mean_fastest_commute",
+#           "dest_hilliness_mean_fastest_commute"
+# 
+# )]
+# make_geojson_zones(z, "outputs/zones.geojson")
+
+make_geojson_zones <- function(z, path = "outputs/zones.geojson"){
```
Comment (Robinlovelace) [inline R/make_geojson.R:27]: Nice function!

File: R/zone_stats.R [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+#' Function to summarise routes into zone statistics
+#' @param r sf data frame of routes returned by cyclestreets
+#' @param route_type name of route type e.g. fastest, quietest
+#' @param route_purose name of route type e.g. commute, school
+#' 
+#' the names are appended to column names 
+routes_to_zone_stats <- function(r, route_type = "fastest", route_purpose = "commute"){
+  
+  # Drop unneeded data
+  r <- sf::st_drop_geometry(r)
```
Comment (Robinlovelace) [inline R/zone_stats.R:11]: Good to drop the geometry before all this group_by summarise stuff :+1:

**Review by Robinlovelace — APPROVED** [review]

**Comment by Robinlovelace** [comment]
Great work @mem48 this is good to merge :rocket:

**Comment by Robinlovelace** [comment]
Couple of nitpicks @mem48 please reply to my comments/questions above. And good practice to delete branches after merging. You should see a "Delete branch" button somewhere.

File: R/make_geojson.R (hunk 3 above)
Comment (mem48) [reply R/make_geojson.R:6]: Wasn't sure this would be in the long-term workflow but needed for current vis

File: .gitignore (hunk 1 above)
Comment (mem48) [reply .gitignore:19]: It's not useful, just gibberish

**Pull request merged**